src/compiler/checker.ts

```diff
@@ -6551,19 +6609,30 @@ module ts {
             }
         }
 
+        function createPromiseType(awaitedType: Type, func: FunctionLikeDeclaration): Type {
+            if (globalPromiseSymbol) {
+                awaitedType = getAwaitedType(awaitedType, awaitedType);
+                var promiseType = globalPromiseType || getDeclaredTypeOfSymbol(globalPromiseSymbol);
+                return promiseType !== emptyObjectType ? createTypeReference(<GenericType>promiseType, [awaitedType]) : emptyObjectType;
+            }
+            error(func, Diagnostics.An_async_function_or_method_must_have_a_valid_awaitable_return_type);
+            return unknownType;
+        }

```

Comment:
I assume elsewhere, you will check that Promise indeed resolves in the value space to the globalPromiseSymbol